### PR TITLE
[Experimental] Add styling controls

### LINKED ISSUE #51787: Product Filters Redesign > Active Filters: Update block styles to match the design
# Description

The Active block informs shoppers which product filters are active and lets them easily clear them. As part of the work to redesign the Product Filters blocks, we need to update the design of the Active filters block to the new design.

# Dependencies

> [!IMPORTANT]
> This issue is blocked by #51641 

# Acceptance Criteria

## Scenario 1: The filter chips match the design in the editor
_**Given**_  a merchant visits a template
_**and**_  adds the Product Filters (Experimental) block to the template
_**When**_  the block is rendered in the editor
**_Then_** the filter chips match the design.

| Current | Expected |
|--------|--------|
| CleanShot 2024-09-30 at 16 52 39@2x | CleanShot 2024-09-30 at 16 51 36@2x | 

---

## Scenario 2: The filter chips match the design on the frontend
_**Given**_  a shopper visits a page that has the Product Filters block added to it
_**When**_ the block is rendered on the frontend
_**Then**_ the filter chips match the design.

| Current | Expected |
|--------|--------|
| CleanShot 2024-09-30 at 16 52 39@2x | CleanShot 2024-09-30 at 16 51 36@2x | 

---

## Scenario 3: The clear button appears below the filter chips in the editor
_**Given**_  a merchant visits a template
_**and**_  adds the Product Filters (Experimental) block to the template
_**When**_ the Active Filters block is rendered in the editor
_**Then**_ the merchant can see the clear button in the canvas and the blocks list view.

| Current | Expected |
|--------|--------|
| CleanShot 2024-09-30 at 16 52 39@2x | CleanShot 2024-09-30 at 16 51 36@2x | 

---

## Scenario 4: The clear button appears below the filter chips on the frontend
_**Given**_ a shopper visits a page that has the Product Filters block added to it
_**When**_ the block is rendered on the frontend
_**and**_ the shopper select some filter options
_**Then**_ the Active Filters block correctly renders the selected filters
_**and**_ the clear button is displayed

| Reference  |
|---------|
| CleanShot 2024-09-30 at 16 51 36@2x | 

---

## Scenario 5: Shoppers can clear all selected filters
_**Given**_ a shopper visits a page that has the Product Filters block added to it
_**and**_ the shopper select some filter options
_**When**_ the Active Filters block correctly renders the selected filters
_**and**_ the shopper clicks on the Clear button
_**Then**_ the Active Filters does not display any selected filters
_**and**_ the Clear button disappears
_**and**_ all filters are reset to an empty state

---

## Scenario 6: The Orientation setting is shown on the block toolbar and the Inspector Controls
_**Given**_ a merchant adds the Product Filters block to a page/template
_**When**_ they select the Product Filters block
_**Then**_  the Orientation setting is shown on the block toolbar [Image Reference 1]
_**and**_  the Orientation setting is shown on the Inspector Control [Image Reference 2]

| Image Reference 1: Orientation settings on the block toolbar |
|--------|
| CleanShot 2024-09-30 at 17 12 34@2x | 

| Image Reference 2: Orientation settings within the Inspector Controls |
|--------|
| CleanShot 2024-09-30 at 17 26 06@2x | 

---

## Scenario 7: Merchants can edit the block orientation from the block toolbar
_**Given**_ a template/page has the Product Filters block
_**and**_ a merchant selected the block
_**When**_ they change the block orientation setting in the block toolbar to vertical
_**Then**_ the filter chips are shown vertically

| Reference |
|--------|
| CleanShot 2024-09-30 at 17 22 29@2x | 

---

## Scenario 8: Merchants can edit the block orientation from the inspector controls
_**Given**_ a template/page has the Product Filters block
_**and**_ a merchant selected the block
_**When**_ they change the block orientation setting in the inspector controls to vertical
_**Then**_ the filter chips are shown vertically

| Reference |
|--------|
| CleanShot 2024-09-30 at 17 22 29@2x | 

---

## Scenario 9: Merchants see the setting to display or hide the Clear all button from the Inspector Controls
_**Given**_ a template/page has the Product Filters block
_**When**_ a merchant select the block
_**and**_ visits the Styles settings for the block
_**Then**_ the setting to enable or hide the Clear all button is shown

| Reference |
|--------|
| CleanShot 2024-09-30 at 17 40 35@2x | 

---

## Scenario 10: Merchants can display or hide the Clear all button from the Inspector Controls
_**Given**_ a template/page has the Product Filters block
_**and**_ a merchant selects the block
_**and**_ visit the Styles settings for the block
_**When**_ they disable the Clear all button setting
_**Then**_ the Clear all button is hidden

---

## Scenario 11: Shoppers don't see the Clear All Button when its setting is disabled
_**Given**_ a shopper visits a page that has the Product Filters block added to it
_**and**_ the shopper select some filter options
_**When**_ the Active Filters block correctly renders the selected filters
_**Then**_ the Clear all button is not displayed

---

## Scenario 12: Merchants can edit the Typography, Dimensions and Border settings 
_**Given**_ a template/page has the Product Filters block
_**and**_ a merchant selects the block
_**When**_ visit the Styles settings for the block
_**Then**_ they can edit the Typography, Dimensions, and Border settings
_**and**_ the changes can be seen in the canvas

---

### PR DESCRIPTION
**Pull request by roykho**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/51787

This PR adds additional styling to the active filter block per design.

- Added Orientation and Justification to the toolbar and inspector controls of the Removable Chips block as discussed p1730901995741429-slack-C05UMTA7W0P
- Added Dimensions and Border controls to the parent Active Filters block.
- We decided to not include the block spacing for now as discussed p1730901995741429-slack-C05UMTA7W0P
- I've decided to not include Typography to the parent Active Filters block because each inner blocks actually has their own Typography controls and some of them are even custom. Both Removable chips and Clear button has their own. This is my own opinion though ref: p1730901995741429-slack-C05UMTA7W0P

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

> [!NOTE]  
> **Pre-requisite**
> - If you're using WC Beta Tester plugin, make sure the `experimental-blocks` feature is enabled (Tools > WCA Test Helper > Features).
> - Make sure to reset Product Catalog Template.

1. Navigate to Product Catalog template (Appearance > Editor > Templates > Product Catalog template.
2. Edit the template.
3. With the block inserter, add Product Filters (Experimental) block to the template.
4. Open the Document Overview (List view) and ensure you see the Product Filter Active block in the new design structure.
5. Select the parent Active Filters block
6. Ensure you see Dimensions and Border controls in the inspector controls.
7. Ensure each of those settings work in the editor and frontend.
8. Select the Removable Chips block.
9. Ensure you see the Orientation (Vertical, Horizontal) settings in the toolbar.
10. Go to the Inspector Controls -> Settings Panel and ensure you also see the Orientation settings there.
11. Try toggling vertical and horizontal settings and watch it change in the editor. Save and view it in the frontend. It should reflect the same as your saved settings.
12. The clear button setting has already been tested in prior PR https://github.com/woocommerce/woocommerce/pull/52268

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Adds additional styling settings to active filter block
</details>

### CODE REVIEW

**Review by dinhtungdu — COMMENTED** [review]
Thanks for working on this. The typo, margin, padding, and border controls are working as expected in my test. But others didn't seem to work probably:

- Justification: this seems to have no effect when I changed the option.
- Orientation: this changes the Orientation of the chips, but I'm not sure it's the correct behavior. The setting belongs to the Active block, changing the orientation should affect the direction of its inner blocks as Row block does.
- Block spacing: this also seems to have no effect.

https://github.com/user-attachments/assets/dde59270-edea-4c02-9599-27734d17f188

**Comment by roykho** [comment]
> Justification: this seems to have no effect when I changed the option.

It is working on my end. Also note it only has affect in vertical. I am not sure how it would look or if it even should function for horizontal.


> Orientation: this changes the Orientation of the chips, but I'm not sure it's the correct behavior. The setting belongs to the Active block, changing the orientation should affect the direction of its inner blocks as Row block does.

I purposely did it this way so that the clear button remains at the bottom per design. Do you think we should make all/any inner blocks change with the orientation settings?

> Block spacing: this also seems to have no effect.

Will look into this one.

**Comment by roykho** [comment]
> Block spacing: this also seems to have no effect.

I think block spacing is actually not needed here right? Because block spacing only works if you have blocks inline?

**Comment by dinhtungdu** [comment]
> Because block spacing only works if you have blocks inline?

I believe not only inline. It's the spacing between inner blocks, like Row and Group blocks are doing.

> I think block spacing is actually not needed here right?

I actually agree. We can add it later if needed. 

> > Orientation: this changes the Orientation of the chips, but I'm not sure it's the correct behavior. The setting belongs to the Active block, changing the orientation should affect the direction of its inner blocks as Row block does.
>
> I purposely did it this way so that the clear button remains at the bottom per design. Do you think we should make all/any inner blocks change with the orientation settings?

What I worry about is: it's possible to add inner blocks to the Active block, so even if no one is doing so, we still need to support that case. I think the orientation setting is better moved to Removable Chips instead. I don't see the need to place inner blocks of Active in a row.

**Comment by dinhtungdu** [comment]
As you mentioned, it's easier to add more features but the opposite is not.

**Review by thealexandrelara — CHANGES_REQUESTED** [review]
Great work! Thank you for working on this. Code looks good! I noticed an issue with the vertical orientation when combined with Centered Justification. Also, could you please add some tests covering these new settings?

✅ Scenario 1: Merchants can modify the Dimensions
- Conditions:
    - A merchant edits the Product Catalog template.
    - The merchant adds the Product Filters (Experimental) block.
    - The merchant selects the Product Filters (Experimental) block.
    - The merchant modifies the Padding, Margin, Border, and Border Radius in the Inspector Controls.
- Expected Outcome: The changes are reflected in the editor and on the frontend.
- Summary: Scenario passed as expected. No issues found.

⚠️  Scenario 2: Merchants can modify the Dimensions
- Conditions:
    - A merchant edits the Product Catalog template.
    - The merchant adds the Product Filters (Experimental) block.
    - The merchant selects the Active Filters block.
    - The merchant modifies the Justification and Orientation in the Inspector Controls.
- Expected Outcome: The changes are reflected in the editor and on the frontend.
- Summary: When setting the Justification to Center and the Orientation to Vertical, the result differs between the editor and the frontend.

| Editor | Frontend |
|--------|--------|
| CleanShot 2024-11-07 at 19 08 24@2x | CleanShot 2024-11-07 at 19 09 42@2x | 

- Another thing I noticed is that when a border has a large border radius and the justification is set to Left or Right, the items get cut off. Although merchants can address this issue by adjusting the padding, it is important to highlight this point:

CleanShot 2024-11-07 at 19 10 19@2x

**Comment by roykho** [comment]
@thealexandrelara Thanks for the review.

> When setting the Justification to Center and the Orientation to Vertical, the result differs between the editor and the frontend.

This is what I see in the editor and frontend.

Editor:


Frontend:


> Another thing I noticed is that when a border has a large border radius and the justification is set to Left or Right, the items get cut off. Although merchants can address this issue by adjusting the padding, it is important to highlight this point:

I believe that's just how GB behaves. You can replicate this with core blocks by adding a group and several inner blocks.


> Also, could you please add some tests covering these new settings?

I am not sure if we should do that as these styling settings are not a critical flow and would only further slow our CIs down. I think only functional items we should add tests for. For example the clear button. If clear button doesn't work on click, that is a significant issue. We should only write tests for those.

**Comment by thealexandrelara** [comment]
This is interesting, I created a new JN site to ensure it's not a cache problem, and for me this issue is still happening . When I access the page containing the Filters block for the first time and select some filter options, the issue occurs. However, the issue does not happen if I select some filter options and reload the page, or if I open a URL that contains filter query parameters that pre-populate the Active Filters block. Here is a video showing this behavior:

https://github.com/user-attachments/assets/4087dec8-7a8d-4600-a076-c93b8359c4eb

> I believe that's just how GB behaves. You can replicate this with core blocks by adding a group and several inner blocks. 

I see, so yeah, that's something I believe we cannot control. The good thing is that merchants can see that in the Editor, so they can make their own adjustments.

> I am not sure if we should do that as these styling settings are not a critical flow and would only further slow our CIs down. I think only functional items we should add tests for. For example the clear button. If clear button doesn't work on click, that is a significant issue. We should only write tests for those.

TBH, I don’t think we should limit our testing just because CI might slow down. This is a separate issue that would require different solutions. Tests are essential for ensuring reliability and preventing regressions. Even if we lose some extra seconds during the process, IMO it is worth it to prevent a bug from reaching production, which would then require debugging, fixing, reviewing, coordinating a patch release, etc., and take significantly longer.

I’m not suggesting visual testing, but at a minimum, we should ensure that these settings are there and can be modified.

**Comment by roykho** [comment]
Thanks for the review @thealexandrelara 

I've added basic test to ensure both layout settings exists and clicking on each of them produces the correct css class names to the container. Though I am still not convince we need to add these tests. Mainly because these are GB core functionality. If we were to add custom settings, yeah I would agree we should add tests. But any how it's there now.

> I created a new JN site to ensure it's not a cache problem, and for me this issue is still happening

~~Yeah I was finally able to reproduce this. I am not entirely sure what the fix would be. My hunch is because the chips are added dynamically, the width is not fully calculated at the time and causes this issue. When you refresh, it works is because the chips are already known (in the URL params) and it knows the calculated width. But again this is just my speculation. I will dig in further to see how this can be remedied but if you have any suggestions, it would be great.~~

Above is fixed.

**Comment by roykho** [comment]
Note: There is a failing e2e tests (the ones added in this PR) ref https://github.com/woocommerce/woocommerce/actions/runs/11759348403/job/32758726253?pr=52598#step:9:253

I am not sure what is causing this error but the tests passes locally.

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterRemovableChips.php [hunk 1]
```diff
@@ -24,33 +24,35 @@ final class ProductFilterRemovableChips extends AbstractBlock {
...
-		$wrapper_attributes = get_block_wrapper_attributes(
-			array(
-				'data-wc-interactive' => wp_json_encode( array( 'namespace' => $this->get_full_block_name() ), JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_QUOT | JSON_HEX_AMP ),
-				'data-wc-key'         => wp_unique_prefixed_id( $this->get_full_block_name() ),
-				'class'               => esc_attr( $classes ),
-				'style'               => esc_attr( $style ),
-			)
+		$wrapper_attributes = array(
+			'data-wc-interactive' => wp_json_encode( array( 'namespace' => $this->get_full_block_name() ), JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_QUOT | JSON_HEX_AMP ),
+			'data-wc-key'         => wp_unique_prefixed_id( $this->get_full_block_name() ),
```
Comment (thealexandrelara) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductFilterRemovableChips.php:43]: TIL. Oh, this is great; I've never used it.

**Review by thealexandrelara — APPROVED** [review]
Thank you for working on this! I ran another round of tests, and everything is working as expected. Amazing work, Roy! 🚀 

✅ Scenario 1: Merchants can modify the Dimensions
- Conditions:
    - A merchant edits the Product Catalog template.
    - The merchant adds the Product Filters (Experimental) block.
    - The merchant selects the Product Filters (Experimental) block.
    - The merchant modifies the Padding, Margin, Border, and Border Radius in the Inspector Controls.
- Expected Outcome: The changes are reflected in the editor and on the frontend.
- Summary: Scenario passed as expected. No issues found.

✅   Scenario 2: Merchants can modify the Dimensions
- Conditions:
    - A merchant edits the Product Catalog template.
    - The merchant adds the Product Filters (Experimental) block.
    - The merchant selects the Active Filters block.
    - The merchant modifies the Justification and Orientation in the Inspector Controls.
- Expected Outcome: The changes are reflected in the editor and on the frontend.
- Summary: Scenario passed as expected. No issues found.

**Pull request merged**